File: toggl-report-front/src/utils/rotulos.ts
import type { Agrupamento } from '../api/tipos';

export const OPCOES_AGRUPAMENTO: { valor: Agrupamento; rotulo: string }[] = [
    { valor: 'descricao', rotulo: 'Por descrição' },
    { valor: 'tag', rotulo: 'Por tag' },
    { valor: 'ambos', rotulo: 'Ambos' },
];

export function rotularAgrupamento(valor: Agrupamento): string {
    return OPCOES_AGRUPAMENTO.find((opcao) => opcao.valor === valor)?.rotulo ?? valor;
}

File: toggl-report-back/TogglReport.Console/Apresentacao/Rotulos.cs
namespace RelatorioToggl.Apresentacao;

public static class Rotulos
{
    public static string Agrupamento(string valor) => valor switch
    {
        "descricao" => "Por descrição",
        "tag" => "Por tag",
        "ambos" => "Ambos",
        _ => valor
    };

    public static string Data(DateTime valor) => valor.ToString("dd/MM/yyyy");
}